test/unit/pudl_environment_test.py

```diff
@@ -0,0 +1,157 @@
+"""Test to see if our environment (PUDL_INPUT/OUTPUT, pudl_settings) is set up properly
+in a variety of situations."""
+
+import os
+from io import StringIO
+
+import pytest
+import yaml
+
+from pudl.workspace.setup import get_settings
+
+
+def test_get_settings_in_test_environment_no_env_vars():
+    if os.getenv("PUDL_OUTPUT"):
+        del os.environ["PUDL_OUTPUT"]
+    if os.getenv("PUDL_INPUT"):
+        del os.environ["PUDL_INPUT"]
+
+    workspace = "/test/whatever"
+    default_settings = {
+        "pudl_in": workspace,
+        "pudl_out": workspace,
+    }
+
+    settings_yaml = StringIO(yaml.dump(default_settings))
+
+    settings = get_settings(yaml_file=settings_yaml, live_dbs=False)
+
+    expected_values = {
+        "pudl_in": f"{workspace}",
+        "pudl_out": f"{workspace}/output",
+        "data_dir": f"{workspace}/data",
+    }
+
+    for key, value in expected_values.items():
+        assert (key, settings[key]) == (key, value)
+
+    assert os.getenv("PUDL_OUTPUT") == f"{default_settings['pudl_out']}/output"
+    assert os.getenv("PUDL_INPUT") == f"{default_settings['pudl_in']}/data"
+
+
+def test_get_settings_in_test_environment_no_env_vars_tmpdir(pudl_out_tmpdir):
+    if os.getenv("PUDL_OUTPUT"):
+        del os.environ["PUDL_OUTPUT"]
+    if os.getenv("PUDL_INPUT"):
+        del os.environ["PUDL_INPUT"]
+
+    workspace = "/test/whatever"
+    default_settings = {
+        "pudl_in": workspace,
+        "pudl_out": workspace,
+    }
+
+    settings_yaml = StringIO(yaml.dump(default_settings))
+
+    settings = get_settings(
+        yaml_file=settings_yaml, live_dbs=False, tmp_dir=pudl_out_tmpdir
+    )
+
+    expected_values = {
+        "pudl_in": f"{workspace}",
+        "pudl_out": f"{pudl_out_tmpdir}/output",
+        "data_dir": f"{workspace}/data",
+    }
+
+    print(settings)
+    for key, value in expected_values.items():
+        assert (key, settings[key]) == (key, value)
+
+    assert os.getenv("PUDL_OUTPUT") == f"{pudl_out_tmpdir}/output"
+    assert os.getenv("PUDL_INPUT") == f"{default_settings['pudl_in']}/data"
+
+
+@pytest.mark.parametrize(
+    "settings_yaml",
+    [
+        None,
+        StringIO(
+            yaml.dump(
+                {
+                    "pudl_in": "/test/workspace",
+                    "pudl_out": "/test/workspace",
+                }
+            )
+        ),
+    ],
+)
+def test_get_settings_in_test_environment_use_env_vars(settings_yaml):
+    workspace = "/test/whatever/from/env"
+    os.environ |= {
+        "PUDL_OUTPUT": f"{workspace}/output",
+        "PUDL_INPUT": f"{workspace}/data",
+    }
+
+    settings = get_settings(yaml_file=settings_yaml, live_dbs=False)
+
+    expected_values = {
+        "pudl_in": f"{workspace}",
+        "pudl_out": f"{workspace}/output",
```

Comment:
If `live_dbs=False` should we have `pudl_out` be a tempdir even if we don't pass in `tmp_dir=something`?

Reply:
Now that we're passing in the in/out dirs directly I think this is a moot point.